pkg/local_object_storage/blobstor/fstree/fstree.go

```diff
@@ -346,6 +379,38 @@ func (t *FSTree) GetBytes(addr oid.Address) ([]byte, error) {
 	return dec, nil
 }
 
+func extractCombinedObject(addr oid.ID, data []byte) ([]byte, error) {
+	const (
+		prefixSize = 1
+		addrSize   = sha256.Size
+		lengthSize = 4
+
+		addrOff   = prefixSize
+		lengthOff = addrOff + addrSize
+		dataOff   = lengthOff + lengthSize
+	)
+
+	var notFound bool
+
+	for len(data) > dataOff && data[0] == combinedPrefix {
+		notFound = true // The file _is_ combined, so the object _must_ be there.
+		var l = binary.BigEndian.Uint32(data[lengthOff:dataOff])
+		if bytes.Compare(data[addrOff:lengthOff], addr[:]) == 0 {
```

Comment:
it was not supposed to get bytes this way... we do not follow our own rules... it is either the rules are bad, or we should use `addr.Encode` here

Reply:
Current code is obviously more efficient, even more importantly it's symmetric to `(*syncBatch).write`, so unless the type changes radically we don't care much about the contents.